src/main.c

```diff
@@ -16,47 +16,59 @@ int main(int argc, char **argv)
     mdsys_t sys;
     double t_start;
 
-    printf("LJMD version %3.1f\n", LJMD_VERSION);
+    #ifdef _MPI
+    MPI_Init(NULL, NULL);
+    #endif
+    fill_mpi_struct(&sys);
+    
+    if (sys.mpirank==0) printf("LJMD version %3.1f\n", LJMD_VERSION);
 
     t_start = wallclock();
 
     /* read input file */
-    add_data(stdin, &line, &restfile, &trajfile, &ergfile, &sys, &nprint);
-    
+    if (sys.mpirank==0) add_data(stdin, &line, &restfile, &trajfile, &ergfile, &sys, &nprint);
+    broadcast_data(&sys);
 
     /* allocate memory */
 
     allocate_mem(&sys);
 
-    /* read restart */
-    fp=fopen(restfile,"r");
-    if(fp) {
-        for (i=0; i<sys.natoms; ++i) {
-            fscanf(fp,"%lf%lf%lf",sys.rx+i, sys.ry+i, sys.rz+i);
-        }
-        for (i=0; i<sys.natoms; ++i) {
-            fscanf(fp,"%lf%lf%lf",sys.vx+i, sys.vy+i, sys.vz+i);
+    if(sys.mpirank==0) {
+        /* read restart */
+        fp=fopen(restfile,"r");
+        if(fp) {
+            for (i=0; i<sys.natoms; ++i) {
+                fscanf(fp,"%lf%lf%lf",sys.rx+i, sys.ry+i, sys.rz+i);
+            }
+            for (i=0; i<sys.natoms; ++i) {
+                fscanf(fp,"%lf%lf%lf",sys.vx+i, sys.vy+i, sys.vz+i);
+            }
+            fclose(fp);
+            azzero(sys.fx, sys.natoms);
+            azzero(sys.fy, sys.natoms);
+            azzero(sys.fz, sys.natoms);
+        } else {
+            perror("cannot read restart file");
+            return 3;
         }
-        fclose(fp);
-    } else {
-        perror("cannot read restart file");
-        return 3;
     }
 
     /* initialize forces and energies.*/
```

Comment:
should we use MPI_Barrier here just so call returns at any process only after all members of the communicator have entered the call.

Reply:
There is a broadcast in the force call, so they will sync there. I dont think the barrier is needed